packages/components/src/components/hds/code-block/index.ts

```diff
@@ -176,15 +182,25 @@ export default class HdsCodeBlock extends Component<HdsCodeBlockSignature> {
   }
 
   @action
-  registerTitleElement(element: HdsCodeBlockTitleSignature['Element']): void {
-    this._titleId = element.id;
+  registerTitleElement(): void {
+    // eslint-disable-next-line ember/no-runloop
+    schedule('afterRender', (): void => {
+      const titleElement = this._element.querySelector(
+        TITLE_ELEMENT_SELECTOR
+      ) as HTMLElement;
```

Comment:
[question] was this the only way? I am a bit nervous about using DOM `querySelector` methods inside controllers, looks a bit contrary to [what we should do](https://docs.google.com/document/d/1HTVhXNtZPB34PiLTQ6a8fsyV8TII6aANm7Nr3S73nLY/edit?tab=t.0#heading=h.3mazrh3lycj2).

@zamoore thoughts?